diff --git a/lib/index.js b/lib/index.js
index 79f672d6465cfa8b66637bbd79a63e3e229bf3ac..dc159808562a0a9399c2358d72ea1076f2dd3182 100644
--- a/lib/index.js
+++ b/lib/index.js
@@ -269,7 +269,13 @@ const De = {
         () => t.start,
         () => t.palette,
         () => t.selectStart,
-        () => t.selectLength
+        () => t.selectLength,
+        () => t.scale,
+        () => t.canvasWidth,
+        () => t.canvasHeight,
+        () => t.rate,
+        () => t.gridRatio,
+        () => t.showShadowText
       ],
       () => {
         k(i);
diff --git a/src/canvas-ruler/index.vue b/src/canvas-ruler/index.vue
index bad91d65b25d283ae429d546a193d79e264c0173..76e662c22eb56c589c40a2e32d3ce71ff5fa1686 100644
--- a/src/canvas-ruler/index.vue
+++ b/src/canvas-ruler/index.vue
@@ -109,7 +109,13 @@ watch(
     () => props.start,
     () => props.palette,
     () => props.selectStart,
-    () => props.selectLength
+    () => props.selectLength,
+    () => props.scale,
+    () => props.canvasWidth,
+    () => props.canvasHeight,
+    () => props.rate,
+    () => props.gridRatio,
+    () => props.showShadowText
   ],
   () => {
     drawRuler(ratioValue)
